### ISSUE TITLE
MovingAverage num_updates support Variable

### ISSUE THREAD

**Issue by fsx950223** [tensorflow/addons#1977]
**Describe the feature and the current behavior/state.**
According to https://github.com/tensorflow/addons/blob/a0bfe3f35f310bdfee53078bdee15e10af342f53/tensorflow_addons/optimizers/moving_average.py#L84
average_decay is a tensor.
If I move it to ```_get_decay```, it could be calculated dynamically with num_updates variable, which aligns with tf1.x ema.

**Relevant information**
- Are you willing to contribute it (yes/no):yes
- Are you willing to maintain it going forward? (yes/no):yes
- Is there a relevant academic paper? (if so, where):
- Is there already an implementation in another framework? (if so, where):
- Was it part of tf.contrib? (if so, where):

**Which API type would this fall under (layer, metric, optimizer, etc.)**
optimizer
**Who will benefit with this feature?**

**Any other info.**

**Comment by facaiy** [comment]
Could you explain the cons and pros for the change?

**Comment by fsx950223** [comment]
SGD with step decay
```python
sgd = tf.keras.optimizers.SGD(0.01)
moving_avg_sgd = tfa.optimizers.MovingAverage(sgd, num_updates = sgd.iterations)
```

**Comment by fsx950223** [comment]
My mistake, I could use dynamic decay.
